test/NuGet.Clients.Tests/NuGet.VisualStudio.Implementation.Test/Extensibility/VsFrameworkCompatibilityTests.cs

```diff
@@ -52,6 +52,24 @@ public void VsFrameworkCompatibility_GetNearestRejectsNullFallbackFrameworks()
             Assert.Throws<ArgumentNullException>(() => target.GetNearest(targetFramework, fallbackTargetFrameworks, frameworks));
         }
 
+        [Fact]
+        public void VsFrameworkCompatibility_GetNearestNet5EraFails()
+        {
+            // Arrange
+            var target = new VsFrameworkCompatibility();
+            var targetFramework = new FrameworkName(".NETCoreApp,Version=v5.0");
+            var frameworks = new[] {
+                new FrameworkName(".NETCoreApp,Version=v4.0"),
+                new FrameworkName(".NETCoreApp,Version=v5.0.0"),
+                new FrameworkName(".NETCoreApp,Version=v5.1.0"),
+                new FrameworkName(".NETCoreApp,Version=v6.0"),
+            };
+
+            // Act
+            Assert.Throws<System.ArgumentException>(() => target.GetNearest(targetFramework, frameworks));
+
```

Comment:
tailing whitespace should be removed.

But also, why isn't `.NETCoreApp,Version=v4.0` compatible with net5? I mean, I know it's not a real .NET Core version, but according to the rule of supporting the same platform identifier with a lower version number, it should be compatible.

Reply:
This is just to verify that we do, indeed, fail on .NET5+ TFMs with this API. This is one of the IVs APIs that need to be updated or have a sibling API created that supports Platforms.

Reply:
(the error it checks for is the argument error for trying to do `new FrameworkName(nearest.DotNetFrameworkName)`)

Reply:
probably not a big deal, but I still don't understand. 

`net4.0` is not compatible with `net5.0`, but that's because `net4.0` == `.NETFramework,Version=4.0`, whereas `net5.0` == `.NETCoreApp,Version=5.0`. In other words the TargetFrameworkIdentifier (TFI) are different, even though the "short folder name" looks the same.

But when the TFI is the same (`.NETCoreApp`), version 5.0 should generally be compatible with lower versions, including 4.0.

If `net5.0` (`.NETCoreApp,Version=5.0`) isn't compatible with `netcoreapp3.1` (`.NETCoreApp,Version=3.1`) then we have a bug that will impact customers.